From ffd069bdf6fdc0ccd295b11978a7ac6dc69c51ef Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Thu, 29 Oct 2026 19:23:03 +0100
Subject: [PATCH] Disable reflection toggle when plugin is unavailable

Disable the reflection toggle when the cubeaddon plugin is unavailable
to avoid presenting an unusable setting.
---
 simple-ccsm.in | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/simple-ccsm.in b/simple-ccsm.in
index d894fdc..c929d2d 100755
--- a/simple-ccsm.in
+++ b/simple-ccsm.in
@@ -984,11 +984,12 @@ class EffectPage:
 
     def SetReflection(self):
         widget = self.Builder.get_object("enableReflection")
+        plugin = self.Context.Plugins.get('cubeaddon')
 
-        if not 'cubeaddon' in self.Context.Plugins:
+        widget.set_sensitive(plugin is not None)
+        if plugin is None:
             return
 
-        plugin = self.Context.Plugins['cubeaddon']
         setting = plugin.Screens[0]['reflection']
         value = setting.Value
         widget.set_active(value)
-- 
GitLab

